include/h2o.h

```diff
@@ -411,6 +417,41 @@ typedef struct st_h2o_mimemap_type_t {
     } data;
 } h2o_mimemap_type_t;
 
+/* errors emanating from h2o */
+enum {
```

Comment:
How about moving the definitions of `H2O_HTTP2_ERROR_*` in http2_internal.h here, instead of trying to define a separate enum?